### PR TITLE
[Zen2] Hide not recovered state

### PR DESCRIPTION
**Pull request by andrershov**
This PR hides `ClusterState`, that has `STATE_NOT_RECOVERED_BLOCK`,
from `ClusterStateAppliers`.
This is needed, because some appliers, such as `IngestService`, rely on
the fact, that cluster state with `STATE_NOT_RECOVERED_BLOCK` won't
contain anything useful.
Once the state is recovered it's fully available for the appliers.

Also, this PR switches most of the tests that require persistence/state
recovery from Zen1 to Zen2.
There are still the following tests, that were assumed to work after
persistence/recovery is implemented, but do not, namely:
1. ~~PrimaryAllocationIT~~ - fixed with https://github.com/elastic/elasticsearch/pull/36224/commits/7d26c64d77d2ce615c328f4c231fca82a97d0aaf
2. GatewayIndexIT.testDanglingIndices and
testIndexDeletionWhenNodeRejoins
3. RemoveCorruptedShardCommandIT

### CODE REVIEW

**Review by ywelsch — CHANGES_REQUESTED** [review]
I've request one small change to the hiding logic, o.w. looking good.

File: modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/IngestRestartIT.java [hunk 1]
```diff
@@ -46,13 +46,6 @@
 @ESIntegTestCase.ClusterScope(numDataNodes = 0, numClientNodes = 0, scope = ESIntegTestCase.Scope.TEST)
 public class IngestRestartIT extends ESIntegTestCase {
 
-    @Override
-    protected Settings nodeSettings(int nodeOrdinal) {
-        return Settings.builder().put(super.nodeSettings(nodeOrdinal))
-            .put(TestZenDiscovery.USE_ZEN2.getKey(), false) // no state persistence yet
```
[the remark this answers is not in the document]
Comment (andrershov) [reply modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/IngestRestartIT.java]: https://github.com/elastic/elasticsearch/pull/36224/commits/0591d54cd60de120fff11a3ef3f69947b17ff6f2

File: x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/MlDistributedFailureIT.java [hunk 2]
```diff
@@ -47,13 +47,6 @@
 
 public class MlDistributedFailureIT extends BaseMlIntegTestCase {
 
-    @Override
-    protected Settings nodeSettings(int nodeOrdinal) {
-        return Settings.builder().put(super.nodeSettings(nodeOrdinal))
-            .put(TestZenDiscovery.USE_ZEN2.getKey(), false) // no state persistence yet
```
[the remark this answers is not in the document]
Comment (andrershov) [reply x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/MlDistributedFailureIT.java]: https://github.com/elastic/elasticsearch/pull/36224/commits/0591d54cd60de120fff11a3ef3f69947b17ff6f2

File: server/src/test/java/org/elasticsearch/cluster/routing/PrimaryAllocationIT.java [hunk 3]
```diff
@@ -341,17 +341,17 @@ public boolean clearData(String nodeName) {
...
         assertAcked(client().admin().indices().prepareCreate("test").setSettings(Settings.builder()
             .put("index.number_of_shards", randomIntBetween(1, 3)).put("index.number_of_replicas", 2)).get());
         ensureGreen("test");
         client().prepareIndex("test", "type1").setSource(jsonBuilder()
             .startObject().field("field", "value1").endObject()).get();
         logger.info("--> removing 2 nodes from cluster");
-        internalCluster().stopRandomDataNode();
-        internalCluster().stopRandomDataNode();
-        internalCluster().fullRestart();
+        internalCluster().stopRandomNode(InternalTestCluster.nameFilter(nodes.get(1)));
```
[the remark this answers is not in the document]
Comment (andrershov) [reply server/src/test/java/org/elasticsearch/cluster/routing/PrimaryAllocationIT.java]: https://github.com/elastic/elasticsearch/pull/36224/commits/254aa2fba23aa23140d54f79908dd848627e834c

**Comment by andrershov** [comment]
@ywelsch I've fixed nits, for some reason I don't see your comments regarding changing hiding logic, can you please comment again?

**Comment by ywelsch** [comment]
weird, my comment has disappeared. I had asked for the hiding logic to keep the coordinationmetadata and the cluster uuid in the empty metadata, as these are particularly useful at this stage where the state might not be recovered yet, but the cluster state is already accessible through the cluster state API.

**Comment by andrershov** [comment]
@ywelsch I've improved hiding logic in https://github.com/elastic/elasticsearch/pull/36224/commits/9c2bfed2d81490dbb2ef09cab3e374dd137bf948, can you please take a look?

**Review by ywelsch — APPROVED** [review]
LGTM

**Pull request merged**